spec/person_spec.rb

```diff
@@ -0,0 +1,23 @@
+require './person'
+
+describe Person do
+  context 'testing Person class' do
+    person = Person.new(name: 'David', age: 20, id: 1, parent_permission: true)
+
+    it 'Person name should be David' do
+      expect(person.name).to eq('David')
+    end
+
+    it 'Person age should be 20' do
+      expect(person.age).to eq(20)
+    end
+
+    it 'Person id should be 1' do
+      expect(person.id).to eq(1)
+    end
+
+    it 'Person parent_permission should be true' do
+      expect(person.parent_permission).to eq(true)
+    end
+  end
```

Comment:
- [ ] Good job with test coverage so, but since we are aiming for a 100% test coverage, it will be nice to cover all our methods in all our classes with tests, for example in this `Person class`, the following tests could be added; 
- Test for the `add_rental method`: 
```
  context "When adding a new rentals" do

    it "The add_rental method returns a rental and adds its to the book" do
      book = Book.new("Title", "Author")
      person = Person.new(24, "Bob")

      rental = person.add_rental(book, "2021-05-04")

      expect(person.rentals).to eql [rental]
    end
  end
```
- The for `checking_permission method`: 
```ruby
  context "When checking permissions" do
    
    it "The can_use_services? method returns false if under age and no parent permission" do
      person = Person.new(14, "Rose", false)
      expect(person.can_use_services?).to be false
    end
    
    it "The can_use_services? method returns true if over age" do
      person = Person.new(19, "Sam", false)
      expect(person.can_use_services?).to be true
    end
    
    it "The can_use_services? method returns true if has parent permission" do
      person = Person.new(14, "Rose", true)
      expect(person.can_use_services?).to be true
    end
  end
```
- Test for `validating_name method`: 
```ruby
  context "When validating name" do

    it "The validate_name method corrects the name" do
      person = Person.new(19, "alongnameiscorrected")
      person.validate_name()
      expect(person.name).to eql "Alongnamei"
    end
  end
```
#### Please it's cover all our tests in all our classes with tests and demonstrated about, it will also be nice to check for positive and negative cases. Checkout this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/rspec_unit_test.md) for more clarification.

Reply:
@t-yanick Thank you for your feedback, Sorry if I miss that because I didn't know that we aiming for 100% coverage in the test, I will work on this fix as soon as possible.